tiledb/sm/c_api/tiledb.cc

```diff
@@ -3974,6 +4056,143 @@ int32_t tiledb_serialize_array_max_buffer_sizes(
   return TILEDB_OK;
 }
 
+capi_return_t tiledb_deserialize_array_delete_fragments_timestamps_request(
```

Comment:
I'm not sure I understand why this function both deserializes and executes the delete fragments request, let's sync to discuss how to simplify this, ideally following the model of something like ` tiledb_deserialize_query_and_array` is simple enough, decently testable and validated in real interactions with rest server.